devnet-sdk/system/interfaces.go

```diff
@@ -28,7 +28,10 @@ type Chain interface {
 	Wallets(ctx context.Context) ([]Wallet, error)
 	ContractsRegistry() interfaces.ContractsRegistry
 	SupportsEIP(ctx context.Context, eip uint64) bool
+	Node() Node
+}
 
+type Node interface {
```

Comment:
genuine question: is this the abstraction level at which gas/nonce should be handled?
I thought those were concepts that applied at the chain level, but I might very well be wrong

My current reading for the approach is it might be valuable to  ask for that (global) information from a specific node, hence the split?

If that's correct, maybe we should have a "chain data provider" interface, that both the chain and the node implement? (so that tests that don't want to care about nodes are free to ignore them and talk to "the chain" directly)

Reply:
My intention of this PR was to decouple the chain from RPC methods, so `Node` will be the "chain data provider". As you mentioned there is a possibility that two "chain data provider" may return different data, but I did not consider this because currently `Chain` receives single RPC url. `Chain` can wrap `Node` which eventually gets its "chain data provider".

Reply:
How would you feel about us having subinterfaces for EL vs CL clients?

Reply:
@teddyknox Sure. we can add EL vs CL clients based on this PR. There must be a use case that we need to invoke CL client APIs, although I could not find the CL API invocation here: https://github.com/ethereum-optimism/optimism/pull/14631. We can revisit them in a separate PR.

Reply:
ditto https://github.com/ethereum-optimism/optimism/pull/14660#discussion_r1983268640

Reply:
sgtm